code/main.py

```diff
@@ -188,6 +115,109 @@ def main():
 
     print("::debug::Successfully finished Azure Machine Learning Train Action")
 
+    wait_for_completion = False
+    # as we don't want to wait here, we just return the run object from here.
+    if parameters.get("wait_for_completion", True):
+        wait_for_completion = True
+
+    return (run, wait_for_completion)
+
+
+def postRun(submittedRuns_for_wait):
+    # Waiting for run to complete
+    print("::debug::Waiting for run to complete")
+    run_pending = True
+
+    while run_pending:
+        tempStack = submittedRuns_for_wait
+        for run in tempStack:
+            if run.get_status() in ['Completed', 'Failed']:
+                # Creating additional outputs of finished run
+                run_metrics = run.get_metrics(recursive=True)
+                run_metrics_markdown = convert_to_markdown(run_metrics)
+                print(f"::set-output name=run_metrics::{run_metrics}")
+                print(f"::set-output name=run_metrics_markdown::{run_metrics_markdown}")
+                submittedRuns_for_wait.remove(run)
+            time.sleep(10)  # wait for 10 seconds to check again.
```

Comment:
this wait time is point of check, do we wait for 10 seconds before checking status again?

Reply:
Given runs can be quite long, this seems reasonable. @marvinbuss thoughts?